Buildings/Fluid/CHPs/BaseClasses/Controller.mo

```diff
@@ -0,0 +1,470 @@
+          -200},{206,-200}}, color={255,0,255}));
+  connect(PEleNet, warUpCtr.PEleNet) annotation (Line(points={{-280,-40},{80,
+          -40},{80,-42},{88,-42}}, color={0,0,127}));
+  connect(PEle, warUpCtr.PEle) annotation (Line(points={{-280,-80},{-40,-80},{
+          -40,-46},{88,-46}}, color={0,0,127}));
+annotation (
+    defaultComponentName="conMai",
+    Diagram(coordinateSystem(extent={{-260,-220},{260,220}})),
+    Icon(coordinateSystem(extent={{-100,-100},{100,100}}), graphics={
+        Polygon(
+          points={{50,34},{50,-18},{90,8},{90,8},{50,34}},
+          lineColor={28,108,200},
+          fillColor={28,108,200},
+          fillPattern=FillPattern.Solid),
+        Ellipse(
+          extent={{-50,40},{10,-22}},
+          lineColor={28,108,200},
+          fillColor={28,108,200},
+          fillPattern=FillPattern.Solid),
+        Line(points={{10,8},{30,8}}, color={0,0,0}),
+        Line(points={{-92,6},{-68,6}}, color={0,0,0}),
+        Line(points={{-20,-44},{-20,-70}}, color={0,0,0}),
+        Polygon(
+          points={{-72,20},{-72,-8},{-50,6},{-72,20}},
+          lineColor={0,0,0},
+          fillColor={215,215,215},
+          fillPattern=FillPattern.Solid),
+        Polygon(
+          points={{28,22},{28,-6},{50,8},{28,22}},
+          lineColor={0,0,0},
+          fillColor={215,215,215},
+          fillPattern=FillPattern.Solid),
+        Polygon(
+          points={{-11,14},{-11,-14},{11,0},{-11,14}},
+          lineColor={0,0,0},
+          fillColor={215,215,215},
+          fillPattern=FillPattern.Solid,
+          origin={-21,-34},
+          rotation=90)}),
+    Documentation(info="<html>
+<p>
+The CHP plant switches between six possible operating modes depending on the current mode,
+control signals and plant boundary conditions. The regular transition between them
+is as follows:
+</p>
+<ul>
+<li>
+off
+</li>
+<li>
+stand-by mode
+</li>
+<li>
+pump-on mode
+</li>
+<li>
+warm-up mode
+</li>
+<li>
+normal operation
+</li>
+<li>
+cool-down mode
+</li>
+<li>
+off or stand-by mode
+</li>
+</ul>
+<h4>Switching between operating modes</h4>
+<p>
+From the off mode
+</p>
+<ul>
+<li>
+The transition from the off to the stand-by mode will occur when the plant availability
+signal <code>avaSig</code> becomes true.
+</li>
+</ul>
+<p>
+From the stand-by mode
+</p>
+<ul>
+<li>
+The transition from the stand-by to the pump-on mode will occur when the plant
+running signal <code>runSig</code> becomes true.
+</li>
+<li>
+If <code>runSig</code> becomes false, the CHP will automatically change to the off mode.
+</li>
+</ul>
+<p>
+From the pump-on mode
+</p>
+<ul>
+<li>
+The transition from the pump-on to stand-by mode will occur after the specified
+time delay and if the water flow rate <code>mWat_flow</code> is greater than
+the minimum <code>mWatMin</code>.
+</li>
+<li>
+If <code>avaSig</code> becomes false or if <code>runSig</code> becomes false,
```

Comment:
The whole logic with `avaSig` and `runSIg` should be revised as in `ThermalElectricalFollowing` the variable `runSig` is actually computed from `avaSig` with an `and` operator.

Reply:
Addressed